src/main/java/fi/otavanopisto/kuntaapi/server/integrations/AbstractIdProvider.java

```diff
@@ -0,0 +1,204 @@
+    return translateId(organizationServiceId, IdType.ORGANIZATION_SERVICE, OrganizationServiceId.class, target);
+  }
+  
+  @Override
+  public EventId translate(EventId eventId, String target) {
+    return translateId(eventId, IdType.EVENT, EventId.class, target);
+  }
+
+  @Override
+  public NewsArticleId translate(NewsArticleId newsArticleId, String target) {
+    return translateId(newsArticleId, IdType.NEWS_ARTICLE, NewsArticleId.class, target);
+  }
+
+  @Override
+  public AttachmentId translate(AttachmentId attachmentId, String target) {
+    return translateId(attachmentId, IdType.ATTACHMENT, AttachmentId.class, target);
+  }
+
+  @Override
+  public BannerId translate(BannerId bannerId, String target) {
+    return translateId(bannerId, IdType.BANNER, BannerId.class, target);
+  }
+
+  @Override
+  public TileId translate(TileId tileId, String target) {
+    return translateId(tileId, IdType.TILE, TileId.class, target);
+  }
+
+  @Override
+  public PageId translate(PageId pageId, String target) {
+    return translateId(pageId, IdType.PAGE, PageId.class, target);
+  }
+
+  @Override
+  public MenuId translate(MenuId menuId, String target) {
+    return translateId(menuId, IdType.MENU, MenuId.class, target);
+  }
+
+  @Override
+  public FileId translate(FileId fileId, String target) {
+    return translateId(fileId, IdType.FILE, FileId.class, target);
+  }
+
+  @Override
+  public MenuItemId translate(MenuItemId menuItemId, String target) {
+    return translateId(menuItemId, IdType.MENU_ITEM, MenuItemId.class, target);
+  }
+
+  @Override
+  public JobId translate(JobId jobId, String target) {
+    return translateId(jobId, IdType.JOB, JobId.class, target);
+  }
+
+  @Override
+  public AnnouncementId translate(AnnouncementId announcementId, String target) {
+    return translateId(announcementId, IdType.ANNOUNCEMENT, AnnouncementId.class, target);
+  }
+
+  private <T extends Id> T translateId(T id, IdType type, Class<T> idClass, String target) {
+    if (!isSupportedType(type)) {
+      return null;
+    }
+    
+    if (id.getSource().equals(target)) {
+      return id;
+    }
+
+    if (getSource().equals(id.getSource()) && KuntaApiConsts.IDENTIFIER_NAME.equals(target)) {
+      return translateToKuntaApiId(id, idClass);
+    } else if (target.equals(getSource()) && KuntaApiConsts.IDENTIFIER_NAME.equals(id.getSource())) {
+      return translateToSourceId(id, type, idClass);
+    }
+    
+    return null;
+  }
+
+  private <T extends Id> T translateToSourceId(T id, IdType type, Class<T> idClass) {
+    Identifier identifier;
+    identifier = identifierController.findIdentifierByTypeSourceAndKuntaApiId(type, getSource(), id.getId());
+    if (identifier != null) {
+      return createId(idClass, getSource(), identifier.getSourceId());
+    }
+    
+    return null;
+  }
+
+  private <T extends Id> T translateToKuntaApiId(T id, Class<T> idClass) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused private "translateToKuntaApiId" method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AUnusedPrivateMethod)